### ISSUE TITLE
King county is not filling their OfficeRecord's OfficeRecordTitle according to CDP's standard

### ISSUE THREAD

**Issue by tohuynh** [CouncilDataProject/cdp-scrapers#66]
### Describe the Bug

_A clear and concise description of the bug._

CDP frontend relies on the number of `Councilmember` roles to calculate a councilmember's terms served since the oldest ingested event. Meaning there should be only one `Councilmember` role per term for a councilmember. This is not the case for some of the councilmembers in King county. See Claudia Balducci's [office records](https://gist.github.com/tohuynh/dd97dda345a3cee0a70b6208a67e389e). In [OfficeRecord 1778](https://gist.github.com/tohuynh/dd97dda345a3cee0a70b6208a67e389e#file-claudia-office-records-json-L95), in which the date range overlaps with her [`Councilmember` role's](https://gist.github.com/tohuynh/dd97dda345a3cee0a70b6208a67e389e#file-claudia-office-records-json-L3) date range, the `OfficeRecordTitle` should be `Member`, not `Councilmember`.

### Expected Behavior

_What did you expect to happen instead?_

I believe OfficeRecordTitle is used as the `role.title` when the `Role` is created.

The eventually created `Role`'s title should be `Member` for OfficeRecord 1778.

---
### Possible solution
Keep track of the "City Council" body name (something to specify what is the main body for the council) for each instance and use it to determine the `Role`'s title. For King County, store "Metropolitan King County Council" somewhere, and whenever there is OffceRecord with that body name, the `Role`'s `title` should be `Councilmember`. If the OfficeRecord doesn't have that body name and the OfficeRecordTitle is `Councilmember`, use OfficeRecordMemberType as `Role`'s title instead. For Seattle, the body name is "City Council".

Basically, perform some sort of post-processing to ensure that the `Role`'s title is in accordance with CDP standards.

cc @dphoria @JacksonMaxfield for additional thoughts or other solutions -- if there's any other way to determine a councilmember's tenure status from their roles.

Alternative: Calculate the tenure status by counting the roles whose `title` ==/contains "Council" (not sure how robust this would be for other instances).

**Comment by JacksonMaxfield** [comment]
Tbh I think this is just a larger issue of us not actually having a standard at all: https://github.com/CouncilDataProject/cdp-backend/blob/main/cdp_backend/database/constants.py

I would be fine with adopting some standards:
```python
class RoleTitles:
    councilmember = "Councilmember"
    president = "President"
    chair = "Chair"
    vice_chair = "Vice Chair"
    alternate = "Alternate"
    member = "Member"
```

And a scraper should use our standard?

**Comment by tohuynh** [comment]
Not sure what the solution should be. If we just use legistar's OfficeRecord without modification, cdp-frontend would incorrectly count OfficeRecord 1778 and others like it as a term.

**Comment by JacksonMaxfield** [comment]
My vote is cast the role value into our of our standard values. cc @BrianL3 for input too.

**Comment by tohuynh** [comment]
> And a scraper should use our standard?
> My vote is cast the role value into our of our standard values

I'm OK with this. Each individual scraper would have to provide their own logic for determining the ingestion_role.title.

**Comment by tohuynh** [comment]
Or we could let cdp-backend take the ingestion_role/OfficeRecord as is and perform the logic when the database `Role` is being created.

**Comment by BrianL3** [comment]
I like Jackson's solution: standardize the data at the point it gets into the DB. The scraping _and_ munging of data is the scraper/backend's area of expertise, and I would prefer to keep the frontend decoupled from that as much as we can.

**Comment by tohuynh** [comment]
> I like Jackson's solution: standardize the data at the point it gets into the DB. The scraping and munging of data is the scraper/backend's area of expertise, and I would prefer to keep the frontend decoupled from that as much as we can.

Agree

> I would be fine with adopting some standards:
> 
> ```python
> class RoleTitles:
>     councilmember = "Councilmember"
>     president = "President"
>     chair = "Chair"
>     vice_chair = "Vice Chair"
>     alternate = "Alternate"
>     member = "Member"
> ```

We should use "Council President" for similarity with "Councilmember"

**Comment by dphoria** [comment]
> Or we could let cdp-backend take the ingestion_role/OfficeRecord as is and perform the logic when the database `Role` is being created.

To keep in line with "everything returned by scraper is in `IngesionModel`", I agree I prefer to setting `Role.title` to a standard in the scraper.

**Comment by dphoria** [comment]
I think I am not fully understanding, there still exists the problem that, for a given person, only 1 `Role` must have `title = RoleTitles.councilmember`?

**Comment by tohuynh** [comment]
Only one councilmember role per term for a person.

**Comment by dphoria** [comment]
cc @Shak2000 for us to keep in mind as we work on `PortlandScraper`. Though, only Shak has done any meaningful work on it thus far. :sweat_smile: I gotta do better!

**Comment by Shak2000** [comment]
> cc @Shak2000 for us to keep in mind as we work on `PortlandScraper`. Though, only Shak has done any meaningful work on it thus far. 😅 I gotta do better!

Thanks a lot Sung, @dphoria! Your feedback was very much appreciated. I couldn't do it without your feedback,
